operator/cmd/mesh/profile-common.go

```diff
@@ -47,8 +47,16 @@ var scope = log.RegisterScope("installer", "installer", 0)
 // ones that are compiled in. If it does, the starting point will be the base and profile YAMLs at that file path.
 // Otherwise it will be the compiled in profile YAMLs.
 // In step 3, the remaining fields in the same user overlay are applied on the resulting profile base.
+
 // The force flag causes validation errors not to abort but only emit log/console warnings.
-func GenerateConfig(inFilenames []string, setOverlayYAML string, force bool, kubeConfig *rest.Config, l *Logger) (string, *v1alpha1.IstioOperatorSpec, error) {
+func GenerateConfig(
+	inFilenames []string,
+	setOverlayYAML string,
+	force bool,
+	includeDefaults bool,
```

Comment:
this function had gotten into a pretty bad state due to doing too many things and after doing a cleanup recently i'd rather avoid this sort of bolt-on functionality because the main function is already hard to understand without these special addons. 
since the display of the derived profile without inherited values is basically just outputting the profile overlay file, could it be implemented that way rather than overloading GenerateConfig?

Reply:
The output should also include any overrides from the command line via `--set`. This is useful to generate new ICP as a delta from a base profile which can then be committed back to SCM.

I think this would be missed if we simplify output the profile overlay file? What is the right place to add this logic to get that desired behavior?

Reply:
I removed the changes to show the profile w/o defaults (profile overlay) from this PR. PTAL.